test/integration/features/cmd-openshift.feature

```diff
@@ -0,0 +1,114 @@
+      etcd [0-9]+\.[0-9]+\.[0-9]+
+      """
+
+  Scenario: Getting address of internal docker registry
+  Prints the host name and port number of the OpenShift registry to the standard output.
+     Given Minishift has state "Running"
+      When executing "minishift openshift registry" succeeds
+      Then stdout should be valid IP with port number
+
+  Scenario: Getting existing service without route
+      When executing "minishift openshift service nodejs-ex" succeeds
+      Then stdout should contain "nodejs-ex"
+       And stdout should not match
+       """
+       ^http:\/\/nodejs-ex-myproject\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.nip\.io
+       """
+
+  Scenario: Getting non-existing service
+  If service does not exist, user gets an empty table.
+      When executing "minishift openshift service not-present" succeeds
+      Then stdout should not contain "not-present"
+
+  Scenario: Getting service from non-existing namespace
+      When executing "minishift openshift service nodejs-ex --namespace does-not-exist" fails
+      Then stderr should contain "Namespace does-not-exist doesn't exist"
+
+  Scenario: Forgotten service name
+      When executing "minishift openshift service --namespace myapp" fails
+      Then stderr should contain "You must specify the name of the service."
+
+  Scenario: User creates route to the service
+      When executing "oc expose svc/nodejs-ex" succeeds
+      Then stdout should contain
+      """
+      route "nodejs-ex" exposed
+      """
+
+  Scenario: Getting existing service with route
+      When executing "minishift openshift service nodejs-ex" succeeds
+      Then stdout should contain "nodejs-ex"
+       And stdout should match
+       """
+       http:\/\/nodejs-ex-myproject\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.nip\.io
+       """
+
+  Scenario: Getting URL of service using --url flag
+      When executing "minishift openshift service nodejs-ex --url" succeeds
+      Then stdout should be valid URL
+
+  Scenario: Seeing configuration of OpenShift master
+  Minishift openshift config view prints YAML configuration of OpenShift cluster.
+  Note: --target=master is default value for minishift openshift config command
+      When executing "minishift openshift config view" succeeds
+      Then stdout should be valid YAML
+
+#  Scenario: Seeing configuration of OpenShift node
+#      When executing "minishift openshift config view --target node" succeeds
+#      Then stdout should be valid YAML#
+
+  Scenario: Setting configuration on OpenShift master
+      When executing »minishift openshift config set --patch '{"assetConfig": {"logoutURL": "http://www.minishift.io"}}'« succeeds
+      Then stdout should contain "Patching OpenShift configuration"
+      
+
+  Scenario: Restarting the OpenShift cluster
+     Given Minishift has state "Running"
+      When executing "minishift openshift restart" succeeds
+      Then stdout should contain "Restarting OpenShift"
```

Comment:
Again, something is missing here. You need a verification that OpenShift really restarted. One can do this by comparing the container ids. They need to change. You would need to run a docker command prior to restart to determine the id of the container. Then you run the restart and get the id again. They need to differ. I am not sure whether we have all the pieces to do this. I guess you can use @gbraad ability to store the outcome of the command as variable. Not quite sure how you go then about comparing.